### PR TITLE
Jest Tests - Creating Exams

### PR DESCRIPTION
**Pull request by akroon3r**
This PR contains all the jest tests for creating exams. To run these tests, navigate to /queue-management/frontend and run `npm test -t bookings-*`. Replace the * in the command with whatever test you would like to run. Tests are found in /queue-management/frontend/src/test.

### CODE REVIEW

**Comment by akroon3r** [comment]
Pushing multiple commits into this PR over time. I'll notify when it's complete.

File: frontend/src/store/index.js [hunk 1]
```diff
@@ -912,21 +912,32 @@ export const store = new Vuex.Store({
     },
 
     clickAddExamSubmit(context, type) {
+      console.log("TYPE", type)
```
Comment (sjrumsby) [inline frontend/src/store/index.js]: Need to remove these console logs

File: frontend/src/store/index.js [hunk 2]
```diff
@@ -912,21 +912,32 @@ export const store = new Vuex.Store({
...
       return new Promise((resolve, reject) => {
         if (type === 'challenger') {
           context.dispatch('postITAChallengerExam').then(() => {
             resolve('success')
-          }).catch(() => { reject('failed') })
+          }).catch((error) => {
+            console.log(error)
```
Comment (sjrumsby) [inline frontend/src/store/index.js]: These ones are fine though

File: frontend/src/test/bookings-group-exam.test.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+      await page.click("#kc-login");
+      await navigationPromise;
+
+      await page.waitForSelector("label.navbar-user");
+    },
+    maxTestCaseTime
+  );
+
+test(
+    "Navigate to Exam Inventory Page",
```
Comment (sjrumsby) [inline frontend/src/test/bookings-group-exam.test.js]: This test seems poorly named. It looks like it's the "Add Group Exam" test?

File: frontend/src/test/bookings-individual-exam.test.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+      await page.click("#kc-login");
+      await navigationPromise;
+
+      await page.waitForSelector("label.navbar-user");
+    },
+    maxTestCaseTime
+  );
+
+  test(
+    "Navigate to Exam Inventory Page",
```
Comment (sjrumsby) [inline frontend/src/test/bookings-individual-exam.test.js]: Same comment for this test case name

File: frontend/src/test/bookings-group-exam.test.js [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+function delay(time) {
+  return new Promise(function(resolve) {
+    setTimeout(resolve, time);
+  });
+}
+
+async function clickNavMenu() {
+  await page.waitForSelector("#nav-dropdown__BV_toggle_");
+  await page.click("#nav-dropdown__BV_toggle_");
+  await delay(100);
```
Comment (sjrumsby) [inline frontend/src/test/bookings-group-exam.test.js]: All of these delays should probably be extracted.

File: frontend/src/store/index.js (hunk 1 above)
Comment (akroon3r) [reply frontend/src/store/index.js]: fixed

File: frontend/src/store/index.js (hunk 2 above)
Comment (akroon3r) [reply frontend/src/store/index.js]: roger that

File: frontend/src/test/bookings-group-exam.test.js (hunk 3 above)
Comment (akroon3r) [reply frontend/src/test/bookings-group-exam.test.js]: fixed

File: frontend/src/test/bookings-individual-exam.test.js (hunk 4 above)
Comment (akroon3r) [reply frontend/src/test/bookings-individual-exam.test.js]: fixed

File: frontend/src/test/bookings-group-exam.test.js (hunk 5 above)
Comment (akroon3r) [reply frontend/src/test/bookings-group-exam.test.js]: Removed all delay function calls.

**Comment by akroon3r** [comment]
Rebased against 10-4 and all changes that were noted have been fixed and pushed up.

**Comment by akroon3r** [comment]
Working on Exam Inventory actions (edit, return etc.) now, commits coming soon.

**Pull request closed without merging**